sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/QueryPlanner.java

```diff
@@ -185,6 +189,30 @@ protected RelRoot planRoot(SqlValidator validator, SqlNode sqlNode) {
     return root;
   }
 
+  /**
+   * Helper function to parse a data type from a string using calcite's full
+   * type definition syntax.
+   */
+  @SneakyThrows
+  public static RelDataType parseDatatype(String datatype) {
```

Comment:
Should this go into CalciteUtil?

Reply:
I'll actually tie it closer to the planner since it can be pretty expensive to run and the planner can hold the state we would need